ndcube/utils/wcs.py

```diff
@@ -477,3 +478,32 @@ def compare_wcs_physical_types(source_wcs, target_wcs):
     target_wcs = get_low_level_wcs(target_wcs, 'target_wcs')
 
     return source_wcs.world_axis_physical_types == target_wcs.world_axis_physical_types
+
+
+def identify_invariant_axes(source_wcs, target_wcs, input_shape):
+    """
+    Performs a pixel to pixel transformation to identify if there are any invariant axes
+    between the given source and target WCS objects.
+
+    Parameters
+    ----------
+    source_wcs: `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
+
+    target_wcs: `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
+
+    input_shape: `tuple`
+        The array shape of the data.
+
+    Returns
+    -------
+    result: `list`
+        A list of booleans denoting whether the axis is invariant or not.
+        Follows the WCS ordering.
+    """
+
+    input_pixel_coords = np.meshgrid(*[np.arange(n) for n in input_shape])
+
+    output_pixel_coords = pixel_to_pixel(source_wcs, target_wcs, *input_pixel_coords)
+
+   return [np.allclose(input_coord, output_coord, atol=1e-6)
```

Comment:
Why atol? It might be worth exposing atol and rtol as kwargs in this function :thinking: